Mage.Sets/src/mage/cards/s/SwordOfForgeAndFrontier.java

```diff
@@ -0,0 +1,54 @@
+package mage.cards.s;
+
+import mage.ObjectColor;
+import mage.abilities.Ability;
+import mage.abilities.common.DealsDamageToAPlayerAttachedTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.common.ExileTopXMayPlayUntilEndOfTurnEffect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.effects.common.continuous.PlayAdditionalLandsControllerEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.abilities.keyword.ProtectionAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author AhmadYProjects
+ */
+public final class SwordOfForgeAndFrontier extends CardImpl {
+
+    public SwordOfForgeAndFrontier(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
+
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // Equipped creature gets +2/+2 and has protection from red and from green.
+        Ability ability = new SimpleStaticAbility(new BoostEquippedEffect(2,2));
+        ability.addEffect(new GainAbilityAttachedEffect(
+                ProtectionAbility.from(ObjectColor.GREEN,ObjectColor.RED), AttachmentType.EQUIPMENT).setText(
+                "and has protection from red and from green"));
+        this.addAbility(ability);
+        // Whenever equipped creature deals combat damage to a player, exile the top two cards of your library. You may play those cards this turn. You may play an additional land this turn.
+        Ability ability2 = new DealsDamageToAPlayerAttachedTriggeredAbility(new PlayAdditionalLandsControllerEffect(1, Duration.EndOfTurn),"equipped creature", false);
```

Comment:
you should switch the two effects here since they're in the opposite order

Reply:
I changed it but why does the order matter?

Reply:
The order of the effects will affect the order in the text on the card. Ideally it should match the actual text on the card.